src/components/card/ComponentDefinitionCard.vue

```diff
@@ -70,6 +75,55 @@ const props = defineProps({
     default: '',
   },
 });
+let shiftX;
+let shiftY;
+
+/**
+ * Handle mouse movement, required for e2e tests.
+ *
+ * @param {MouseEvent} event - The mouse move event.
+ */
+function onMouseMove(event) {
+  const card = document.getElementById(`component-definition-${props.definition.type}`);
+
+  card.style.left = `${event.pageX - shiftX}px`;
+  card.style.top = `${event.pageY - shiftY}px`;
+}
+
+/**
+ * Setup drag data and add mousemouve handler.
+ *
+ * @param {DragEvent} event - The starting drag event.
+ */
+function dragStartHandler(event) {
+  const dragData = {
```

Comment:
why you don't set directly template or definition in the data:
```
const dragData = {
  isTemplate: props.definition.isTemplate,
  template, // if isTemplate is true,
  definition, // if isTemplate is false
};
```

Reply:
``dragData`` will be converted to a JSON string before begin embedded into ``event.dataTransfer``. It is much less error-prone to pass just the key and fetch the appropriate element on drop than to try to rebuild the class instance.